ipa-core/src/bin/report_collector.rs

```diff
@@ -171,25 +203,150 @@ fn gen_inputs(
     query_style: IpaQueryStyle,
 ) -> Result<(), Box<dyn Error>> {
-    let input = InputSource::from(&args.input);
-    let query_type: QueryType;
-    match (security_model, &query_style) {
-        (IpaSecurityModel::SemiHonest, IpaQueryStyle::Oprf) => {
-            query_type = QueryType::OprfIpaRelaxedDpPadding(ipa_query_config);
+    let query_type = get_query_type(security_model, &query_style, ipa_query_config);
+
+    let enc1 = &args
+        .enc_input_file1
+        .as_ref()
+        .expect("--enc_input_file1 is required");
+    let enc2 = &args
+        .enc_input_file1
+        .as_ref()
+        .expect("--enc_input_file2 is required");
+    let enc3 = &args
+        .enc_input_file1
+        .as_ref()
+        .expect("--enc_input_file3 is required");
+
+    let mut buffers: [_; 3] = std::array::from_fn(|_| Vec::new());
+    for (i, path) in [enc1, enc2, enc3].iter().enumerate() {
+        let file = File::open(path).unwrap_or_else(|e| panic!("unable to open file {path:?}. {e}"));
+        let reader = BufReader::new(file);
+        for line in reader.lines() {
+            let line = line.unwrap();
+            let encrypted_report_bytes = hex::decode(line.trim()).unwrap();
+            buffers[i].put_u16_le(encrypted_report_bytes.len().try_into().unwrap());
+            buffers[i].put_slice(encrypted_report_bytes.as_slice());
         }
-        (IpaSecurityModel::Malicious, IpaQueryStyle::Oprf) => {
-            panic!("OPRF for malicious is not implemented as yet")
+    }
+    // Panic if input sizes are not the same
+    assert!(buffers[0].len() == buffers[1].len());
+    assert!(buffers[1].len() == buffers[2].len());
+
+    // without loss of generality, set query length to length of first input size
+    let query_size = buffers[0].len();
+
+    let query_config = QueryConfig {
+        size: QuerySize::try_from(query_size).unwrap(),
+        field_type: FieldType::Fp32BitPrime,
+        query_type,
+    };
+
+    let query_id = helper_clients[0]
+        .create_query(query_config)
+        .await
+        .expect("Unable to create query!");
+
+    let inputs = buffers.map(BodyStream::from);
+    tracing::info!("Starting query for OPRF");
+    let actual = match query_style {
+        IpaQueryStyle::Oprf => {
+            // the value for histogram values (BA32) must be kept in sync with the server-side
+            // implementation, otherwise a runtime reconstruct error will be generated.
+            // see ipa-core/src/query/executor.rs
+            run_query_and_validate::<BA32>(
+                inputs,
+                query_size,
+                helper_clients,
+                query_id,
+                ipa_query_config,
+            )
+            .await
         }
     };
 
+    if let Some(ref path) = args.output_file {
+        let _ = write_ipa_output_file(path, &actual);
+    } else {
+        println!("{}", serde_json::to_string_pretty(&actual)?);
+    }
+    Ok(())
```

Comment:
this code is all red - sometimes code coverage gives us false negatives. Can you please check that we indeed are executing it (I usually just insert a panic and check that test fails)

Reply:
does code coverage catch stuff covered by the integration tests?

Reply:
it should, `coverage-ci` should include integration tests

Reply:
now that the tests work, it shows as covered!